File: LAB/Lab_1/Z2.scala
def hits[A](listA: List[A], listB: List[A]): Int = {
    if(listA.isEmpty || listB.isEmpty) {
        0
    } else {
        if (listA.head == listB.head) {
            1 + hits(listA.tail, listB.tail);
        } else {
            hits(listA.tail, listB.tail);
        }
    }
}

@main def main() = {
    val listA = List(3, 2, 1);
    val listB = List(1, 2, 3);

    println(hits(listA, listB));

    val listC = List(3, 2, 1);
    val listD = List(1, 2, 1, 0, 0, 0, 0);

    println(hits(listC, listD));

    val listE = List();
    val listF = List(1, 2, 1, 0, 0, 0, 0);

    println(hits(listE, listF));
}

File: LAB/Lab_1/Z2.ml
let rec hits listA listB =
  if listA = [] || listB = [] then
    0
  else
    let headA = List.hd listA in
    let tailA = List.tl listA in
    let headB = List.hd listB in
    let tailB = List.tl listB in
    if headA = headB then
      1 + hits tailA tailB
    else 
      hits tailA tailB

let resultA = hits [3; 2; 1] [1; 2; 3];;
let resultB = hits [3; 2; 1] [1; 2; 1; 0; 0; 0; 0];;
let resultC = hits [] [1; 2; 1; 0; 0; 0; 0];;

Printf.printf "%d, %d, %d" resultA resultB resultC;;